src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -0,0 +1,280 @@
+    {
+        private const string HubName = "HubName";
+        private const string TaskHubName = "taskHubName";
+        private const string Connection = "connection";
+        private const string DurableTaskStorageConnectionName = "azureStorageConnectionStringName";
+        private const string DurableTask = "durableTask";
+
+        private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
+        private readonly ILogger _logger;
+        private readonly IEnumerable<WorkerConfig> _workerConfigs;
+        private readonly HttpClient _httpClient;
+        private readonly ISecretManagerProvider _secretManagerProvider;
+        private readonly IConfiguration _configuration;
+        private readonly IHostIdProvider _hostIdProvider;
+
+        private CloudBlockBlob _hashBlob;
+
+        public FunctionsSyncManager(IConfiguration configuration, IHostIdProvider hostIdProvider, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IOptions<LanguageWorkerOptions> languageWorkerOptions, ILoggerFactory loggerFactory, HttpClient httpClient, ISecretManagerProvider secretManagerProvider)
+        {
+            _applicationHostOptions = applicationHostOptions;
+            _logger = loggerFactory?.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _httpClient = httpClient;
+            _secretManagerProvider = secretManagerProvider;
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+        }
+
+        private CloudBlockBlob HashBlob
+        {
+            get
+            {
+                if (_hashBlob == null)
+                {
+                    string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+                    CloudStorageAccount account = null;
+                    if (!string.IsNullOrEmpty(storageConnectionString) &&
+                        CloudStorageAccount.TryParse(storageConnectionString, out account))
+                    {
+                        string hostId = _hostIdProvider.GetHostIdAsync(CancellationToken.None).GetAwaiter().GetResult();
+                        CloudBlobClient blobClient = account.CreateCloudBlobClient();
+                        var blobContainer = blobClient.GetContainerReference(ScriptConstants.AzureWebJobsHostsContainerName);
+                        string hashBlobPath = $"synctriggers/{hostId}/last";
+                        _hashBlob = blobContainer.GetBlockBlobReference(hashBlobPath);
+                    }
+                }
+                return _hashBlob;
+            }
+        }
+
+        public async Task<(bool success, string error)> TrySyncTriggersAsync(bool checkHash = false)
+        {
+            var payload = await GetSyncTriggersPayload();
+            string json = JsonConvert.SerializeObject(payload);
+
```

Comment:
One of our detector proposals was to log the trigger JSON on startup (the same way ScaleController does today) so that we could more easily compare if the host and ScaleController were out-of-sync. The detector would look in Kusto at the ScaleController and Functions logs and be able to see if they differed. 

Since you're in here now -- would it make sense to write a debug log with this payload? I'm assuming it's the same one the ScaleController logs show us?